TST: two tests from  `astropy/io/registry/tests/test_registries.py` that cannot be run in isolation

This test cannot be run on its own at the moment:
```
$ pytest astropy/io/registry/tests/test_registries.py::TestUnifiedOutputRegistry::test_delay_doc_updates
...
FAILED astropy/io/registry/tests/test_registries.py::TestUnifiedOutputRegistry::test_delay_doc_updates - UnboundLocalError: cannot access local variable 'ihd' where it is not associated with a value
```
this test actually contains some logic to workaround test pollution, but it turns out it really *depends* on test pollution to run properly.

--- a/astropy/io/registry/tests/test_registries.py
+++ b/astropy/io/registry/tests/test_registries.py
@@ -353,6 +353,7 @@ class TestUnifiedInputRegistry(TestUnifiedIORegistryBase):
                     assert docs[-1][iread : iread + 3] != "Yes"
         # now test it's updated
         docs = EmptyData.read.__doc__.split("\n")
+        ihd = [i for i, s in enumerate(docs) if ("Format" in s)][0]
         ifmt = docs[ihd].index("Format") + 2
         iread = docs[ihd].index("Read") + 1
         assert docs[-2][ifmt : ifmt + 4] == "test"
@@ -722,6 +723,7 @@ class TestUnifiedOutputRegistry(TestUnifiedIORegistryBase):
                     assert docs[-1][iwrite : iwrite + 3] != "Yes"
         # now test it's updated
         docs = EmptyData.write.__doc__.split("\n")
+        ihd = [i for i, s in enumerate(docs) if ("Format" in s)][0]
         ifmt = docs[ihd].index("Format") + 1
         iwrite = docs[ihd].index("Write") + 2
         assert fmt in docs[-2][ifmt : ifmt + len(fmt) + 1]

